akka-http/src/main/scala/akka/http/javadsl/server/directives/FileAndResourceDirectives.scala

```diff
@@ -17,112 +25,229 @@ trait ContentTypeResolver {
   def resolve(fileName: String): ContentType
 }
 
-/**
- * A resolver that assumes the given constant [[ContentType]] for all files.
- */
-case class StaticContentTypeResolver(contentType: ContentType) extends ContentTypeResolver {
-  def resolve(fileName: String): ContentType = contentType
+// TODO Javify DirectoryListing
+trait DirectoryRenderer {
+  def marshaller(renderVanityFooter: Boolean): Marshaller[DirectoryListing, RequestEntity]
+}
+
+object FileAndResourceDirectives {
+  private[FileAndResourceDirectives] implicit def scalaResolver(resolver: ContentTypeResolver) =
+    new scaladsl.server.directives.ContentTypeResolver {
+      override def apply(fileName: String) = resolver.resolve(fileName)
+    }
+
+  private[FileAndResourceDirectives] implicit def scalaRenderer(renderer: DirectoryRenderer) =
```

Comment:
- `private[http]` is enough
- `private[X]` does not affect Java, in fact it makes it `public` in Java (sic), so all cases where we do that need a comment with `/** INTERNAL API */` (so people can't complain if we remove them - it is internal api